crates/aranya-util/src/ready.rs

```diff
@@ -0,0 +1,59 @@
+use std::sync::{
+    atomic::{AtomicUsize, Ordering},
+    Arc,
+};
+
+/// All [`Notifier`]s were dropped before notifying.
+#[derive(Copy, Clone, Debug, Default, Hash, PartialEq, Eq, PartialOrd, Ord, thiserror::Error)]
+#[error("ready notifiers dropped before notifying")]
+pub struct WaitError;
+
+/// Waits for `n` tasks to be ready.
+#[derive(Debug)]
+pub struct Waiter {
+    rx: tokio::sync::mpsc::Receiver<()>,
+    notifier: Notifier,
+}
+
+impl Waiter {
+    /// Create a waiter that will wait for `count` ready notifications.
+    pub fn new(count: usize) -> Self {
+        let (tx, rx) = tokio::sync::mpsc::channel(1);
```

Comment:
What if there are multiple notifications at the same time? Shouldn't the size of the `mpsc` channel be `count`?
```suggestion
        let (tx, rx) = tokio::sync::mpsc::channel(count);
```

Reply:
It's a one-time operation at startup so it doesn't seem like it's worth optimizing yet.